vendors/microchip/boards/secure_element/CMakeLists.txt

```diff
@@ -0,0 +1,202 @@
+)
+
+target_compile_options(
+    AFR::compiler::mcu_port
+    INTERFACE "/MP" "/wd4210" "/wd4127" "/wd4244" "/wd4310"
+)
+
+# -------------------------------------------------------------------------------------------------
+# Amazon FreeRTOS portable layers
+# -------------------------------------------------------------------------------------------------
+# Normally the portable layer for kernel should be vendor's driver code.
+afr_mcu_port(kernel)
+
+# ECC608 source files
+afr_glob_src(ecc608_src DIRECTORY ${ecc608_dir}/lib)
+afr_glob_src(ecc608_atacert_src RECURSE DIRECTORY ${ecc608_dir}/lib/atcacert)
+afr_glob_src(ecc608_basic_src RECURSE DIRECTORY ${ecc608_dir}/lib/basic)
+afr_glob_src(ecc608_crypto_src RECURSE DIRECTORY ${ecc608_dir}/lib/crypto)
+afr_glob_src(ecc608_host_src RECURSE DIRECTORY ${ecc608_dir}/lib/host)
+afr_glob_src(ecc608_pkcs11_src RECURSE DIRECTORY ${ecc608_dir}/lib/pkcs11)
+
+set(
+    ecc608_hid_src
+    "${ecc608_dir}/third_party/hidapi/windows/hid.c"
+)
+set(
+    ecc608_hal_src
+        "${ecc608_dir}/lib/hal/atca_hal.c"
+        "${ecc608_dir}/lib/hal/hal_all_platforms_kit_hidapi.c"
+        "${ecc608_dir}/lib/hal/hal_freertos.c"
+        "${ecc608_dir}/lib/hal/hal_windows.c"
+        "${ecc608_dir}/lib/hal/kit_protocol.c"
+        "${ecc608_dir}/lib/hal/atca_hal.h"
+        "${ecc608_dir}/lib/hal/hal_all_platforms_kit_hidapi.h"
+        "${ecc608_dir}/lib/hal/kit_phy.h"
+        "${ecc608_dir}/lib/hal/kit_protocol.h"
+)
+target_sources(
+    AFR::kernel::mcu_port
+    INTERFACE
+        "${AFR_KERNEL_DIR}/portable/MSVC-MingW/port.c"
+        "${AFR_KERNEL_DIR}/portable/MSVC-MingW/portmacro.h"
+        "${AFR_KERNEL_DIR}/portable/MemMang/heap_4.c"
+        "${ecc608_src}"
+        "${ecc608_atacert_src}"
+        "${ecc608_basic_src}"
+        "${ecc608_crypto_src}"
+        "${ecc608_hal_src}"
+        "${ecc608_hid_src}"
+        "${ecc608_host_src}"
+)
+target_include_directories(
+    AFR::kernel::mcu_port
+    INTERFACE
+        "${AFR_KERNEL_DIR}/portable/MSVC-MingW"
+        "${board_dir}/config_files"
+        "${board_dir}/application_code"
+        "${ecc608_dir}/lib"
+        "${ecc608_dir}/third_party/hidapi/hidapi"
+        # Need aws_clientcredential.h
+        "$<IF:${AFR_IS_TESTING},${AFR_TESTS_DIR},${AFR_DEMOS_DIR}>/include"
+)
+target_link_libraries(
+    AFR::kernel::mcu_port
+    INTERFACE
+        3rdparty::tracealyzer_recorder
+)
+
+# POSIX
+afr_mcu_port(posix)
+target_sources(
+    AFR::posix::mcu_port
+    INTERFACE "${afr_ports_dir}/posix/FreeRTOS_POSIX_portable.h"
+)
+target_include_directories(
+    AFR::posix::mcu_port
+    INTERFACE "${afr_ports_dir}/posix"
+)
+target_link_libraries(
+    AFR::posix::mcu_port
+    INTERFACE AFR::freertos_plus_posix
+)
+
+# PKCS11
+afr_mcu_port(pkcs11_implementation)
+target_sources(
+    AFR::pkcs11_implementation::mcu_port
+    INTERFACE
+        "${ecc608_pkcs11_src}"
+        "${afr_ports_dir}/pkcs11/iot_pkcs11_pal.c"
```

Comment:
Let's get rid of iot_pkcs11_pal.c for this port.

Reply:
done